flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCUpsertOutputFormat.java

```diff
@@ -166,6 +166,22 @@ public synchronized void flush() throws Exception {
 				if (i >= maxRetryTimes) {
 					throw e;
 				}
+				boolean isValid = false;
+				try {
+					isValid = connection.isValid(1);
+				} catch (SQLException ex) {
+					LOG.warn("Check connection valid failed.", ex);
+				}
+
+				try {
+					if (!isValid) {
+						LOG.warn("This connection is not valid, trying to reconnect.");
+						establishConnection();
```

Comment:
Is it possible to put the reconnect operation in the exception handling of isValid?
if reconnect failed,then throw exception and retrying may be meaningless?